solr/core/src/java/org/apache/solr/search/SurroundQParserPlugin.java

```diff
@@ -77,6 +106,11 @@ public Query parse() throws SyntaxError {
           this.maxBasicQueries = DEFMAXBASICQUERIES;
         }
       }
+
+      if (MAX_BASIC_QUERIES_OVERRIDE > 0) {
+        this.maxBasicQueries = MAX_BASIC_QUERIES_OVERRIDE;
+      }
+
```

Comment:
Is it preferable to have the sysprop completely override user-supplied localParam values? My understanding is that we want this to be an absolute ceiling on configured vals -- not to be an absolute override. i.e., query should still be able to specify a _lower_ mbq setting, right?

Reply:
Thank you for the question!

Yes that was the previous implementation (absolute max). But as discussed with @hiteshk25 , the previous implementation would have required re-deployment on all nodes since with such approach we would need to:
1. Set absolute max override for all nodes as 10k, except selected nodes with 50k (and higher solr.max.booleanClauses)
2. Redeploy all node so all nodes so they can process the absolute max 
3. Set higher `maxBasicQueries` from client side

While with the current approach:
1. Set higher `solr.maxBasicQueriesOverride` (and higher solr.max.booleanClauses) only on selected nodes
2. Redeploy on only selected nodes so they can process the absolute max
3. Set higher `maxBasicQueries` from client side

So the current approach has major pro of not needing to restart all nodes (and the code logic is slightly simpler). The con is that we can no longer control is per org and such override basically just ignore all `maxBasicQueries` param from surround query. 

The approach now meant to be a temporary measure, I don't think we want to keep this around at all. We might even want to revert this commit later on since such system prop does not make sense in any normal scenario (unlike the absolute max, which could be useful in some normal scenarios)

Reply:
🤦 of course! sorry I overlooked the discussion that resulted in going this route, and thanks for the explanation.